screenshots/Screenshots.ts

```diff
@@ -1,48 +1,128 @@
+import childProcess from 'child_process';
+import fs from 'fs';
 import os from 'os';
+import path from 'path';
 
 import { expect } from '@playwright/test';
+import dayjs from 'dayjs';
 
 import { NavPage } from '../e2e/pages/nav-page';
 import { PreferencesPage } from '../e2e/pages/preferences';
 
-import type { Page, PageScreenshotOptions } from '@playwright/test';
+import type { Page } from '@playwright/test';
 
 interface ScreenshotsOptions {
   directory?: string;
+  isOsCommand?: boolean;
 }
 
 export class Screenshots {
+  private isOsCommand = true;
+  private sleepDuration = Number(process.env.RD_ENV_SCREENSHOT_SLEEP) || 1000;
+
+  // used by Mac api
+  private appBundleTitle = 'Electron';
+
+  protected windowTitle = '';
   private screenshotIndex = 0;
   readonly page: Page;
   readonly directory: string | undefined;
+  readonly screenshotDirectory: string | undefined;
 
   constructor(page: Page, opt?: ScreenshotsOptions) {
```

Comment:
Sorry to be a pain, but the last round of changes have highlighted a bit of an issue with the `Screenshots` class. I think we should make `directory` a required input. This improves things in a few ways:
- The code that sets `this.directory` and `this.screenshotDirectory` in `constructor` doesn't have to check whether `opts?.directory` is set.
- I think we could remove `Screenshots.directory` altogether - I can't see anything that uses it beyond `Screenshots.constructor`.
- `Screenshots.buildPath` doesn't have to check whether `this.screenshotDirectory` is set.
- `Screenshots.createScreenshotsDirectory` doesn't have to check whether `this.screenshotDirectory` is set.
- We avoid errors/nonsense inputs in the two places where `Screenshots.buildPath` is used in `Screenshots.screenshot`.

Reply:
If you don't want to change this, we should think about changing what `Screenshots.buildPath` returns. As it is, in the `!this.screenshotDirectory` case, it returns `''`. Then, in `this.page.screenshot(options)` in `Screenshots.screenshot`, `options.path` is set to `''`, which might not get handled correctly. We'd want it to be `null` or `undefined`.

Reply:
I made some changes, let me know what you think.

Reply:
:ok_hand: :ok_hand: :ok_hand: